src/lib/y2storage/disk_size.rb

```diff
@@ -118,40 +96,62 @@ def zero
       # Examples:
       #   42 GiB
       #   42.00  GiB
+      #   42 GB
+      #   42GB
       #   512
       #   0.5 YiB (512 ZiB)
+      #   1024MiB(1 GiB)
       #   unlimited
       #
-      # Invalid:
-      #   42 GB    (supporting binary units only)
-      #
-      def parse(str)
-        # ignore everything added in parentheses, so we can also parse the output of #to_s
-        str.gsub!(/\(.*/, "")
-        str.strip!
+      def parse(str, legacy_units: false)
+        str = sanitize(str)
         return DiskSize.unlimited if str == UNLIMITED
-        size_str, unit = str.split(/\s+/)
-        raise ArgumentError, "Bad number: #{size_str}" if size_str !~ /^\d+\.?\d*$/
-        size = size_str.to_f
-        return DiskSize.new(size) if unit.nil?
-        DiskSize.new(size * unit_multiplier(unit))
+        DiskSize.new str_to_bytes(str, legacy_units: legacy_units)
```

Comment:
According to SUSE's style guide, the parenthesis should not be omitted. https://github.com/SUSE/style-guides/blob/master/Ruby.md#methods-1